pkg/jsonutil/conversion_test.go

```diff
@@ -99,5 +131,5 @@ func verifyJSONToProtoToJSON(t *testing.T, inputJSON string, shouldPreserve bool
 	// Conversion to only JSON escapes characters if specified in options.
 	convertedJSON, err := ProtoToJSON(&proto, options...)
 	assert.NoError(t, err)
-	assert.Equalf(t, shouldPreserve, inputJSON == convertedJSON, "original JSON:\n%s\nconvertedJSON:\n%s", inputJSON, convertedJSON)
+	assert.Equalf(t, shouldPreserve, targetJSON == convertedJSON, "original JSON:\n%s\ntargetJSON:\n%s\nconvertedJSON:\n%s", inputJSON, targetJSON, convertedJSON)
```

Comment:
The whole `addUnknownField` thing adds great deal of uncertainty here: is `targetJSON != convertedJSON` because of escaping special characters or did it have something to do with extra `unknownfield` attribute?

In fact, the effect of `unknownfield` attribute with the change is just that `JSONToProto()` call doesn't return error.

One can argue that the tests are still understandable (if you spend enough time reading), but I would object that it becomes harder to maintain these tests as they now try checking two things at once.

I suggest undoing all current changes to this `pkg/jsonutil/conversion_test.go` file and introducing a new simple and boring test (I have not tried compiling this):

```go
func TestNoErrorOnUnknownAttribute(t *testing.T) {
	const json = `{ "id": "6500", "unknownField": "junk" }`
	var proto v1.ResourceByID

	err := JSONToProto(json, &proto)

	assert.NoError(t, err)
	assert.Equal(t, "6500", proto.GetID())
}
```

Reply:
This approach was actually my first thought but I clearly decided to over complicate things and try to fit it in the existing tests.  I will update accordingly.